perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -2428,6 +2480,13 @@ public MemberGroupStatus getUnifiedMemberGroupStatus(PerunSession sess, User use
 		return getMembersManagerImpl().findMembers(sess, vo, searchString, onlySponsored);
 	}
 
+	@Override
+	public void updateSponsorshipValidity(PerunSession sess, Member sponsoredMember, User sponsor,
```

Comment:
We have agreed with @stavamichal on keeping this behavior since we allow this in the membership expiration. If we wanted to change it, we would create a separate PR.